### PR TITLE
Immutable time docs

### PR DESCRIPTION
**Pull request by markstory**
I'm not sure I have everything covered, but if there is more people would like to see I am happy to add more detail.

### CODE REVIEW

**Comment by antograssiot** [comment]
nice work @markstory 
It might be good to mention at least in the migration guide that Chronos requires php 5.5.8 as a minimum but I could make an other Pr on the 3.2 branch to update every places where php 5.4.16 is mentionned maybe..

**Comment by markstory** [comment]
@antograssiot I think I got all the version numbers outside of the 3.0-migration guide. That document can continue saying 5.4 as that is what is needed for that release.

**Comment by antograssiot** [comment]
@markstory great. Feel free to merge or wait for any other feedback, it looks good to go for me

**Comment by lorenzo** [comment]
@markstory do you think having a chapter with the full documentation for chronos is a good thing?
One of the goals that I had in mind for chronos was that it was an excuse for documenting all methods in the book.

**Comment by markstory** [comment]
@lorenzo We could. It has a mountain of methods. Would you want to include examples of each method?

**Comment by ravage84** [comment]
Examples for the most important use cases for chronos are a good start.

File: en/appendices/3-2-migration-guide.rst [hunk 1]
```diff
@@ -11,12 +11,46 @@ CakePHP 3.2 requires at least PHP 5.5. By adopting PHP 5.5 we can provide better
 Date and Time libraries and remove dependencies on password compatiblity
 librarires.
 
+Carbon Replaced with Chronos
+============================
+
+The Carbon library has been replaced with cakephp/chronos. This new library is
```
Comment (ravage84) [inline en/appendices/3-2-migration-guide.rst]: I feel like ``cakephp/chronos`` should link to somewhere. Not sure if it should link to the GH repo or time.rst, though.

File: en/contributing/code.rst [hunk 2]
```diff
@@ -12,7 +12,7 @@ Before working on patches for CakePHP, it's a good idea to get your environment
 setup. You'll need the following software:
 
 * Git
-* PHP 5.4.16 or greater
+* PHP 5.5.8 or greater
```
Comment (ravage84) [inline en/contributing/code.rst]: Just curious, could we repalce the literal for the required PHP version with a constant or variable?

Comment (markstory) [inline en/contributing/code.rst]: Sphinx doesn't have variables from what I remember.

File: en/appendices/3-2-migration-guide.rst (hunk 1 above)
Comment (markstory) [inline en/appendices/3-2-migration-guide.rst]: I can link to the github repo.

**Comment by lorenzo** [comment]
@markstory Maybe examples of some of them. For example we could explain `addDay()` and only mention `subDay()` as it works the same. But I think it is important to make the method names searchable in that way

File: en/appendices/3-2-migration-guide.rst [hunk 3]
```diff
@@ -11,12 +11,46 @@ CakePHP 3.2 requires at least PHP 5.5. By adopting PHP 5.5 we can provide better
...
+New Immutable Date and Time Objects
+===================================
+
+The ``FrozenTime``, and ``FrozenDate`` classes were added. These classes offer
+the same API as the ``Time`` object has. The frozen classes provide immutable
+variants of ``Time`` and ``Date``.  By using immutable objects, you can prevent
+accidental mutations. Instead of in-place modifications, modifier methods return
+*new* instances::
+
+    use Cake\I18\FrozenTime;
```
Comment (Xety) [inline en/appendices/3-2-migration-guide.rst]: Typo : `I18` => `I18n`

Comment (markstory) [inline en/appendices/3-2-migration-guide.rst]: Thanks.

**Comment by markstory** [comment]
@lorenzo Ok, could that be a separate pull request?

**Comment by markstory** [comment]
I'll add examples of the Chronos API in a separate pull request.

**Pull request merged**